File: jscomp/test/webpack_config.js
'use strict';

var List = require("../../lib/js/list.js");
var List$1 = require("List");
var List$2 = require("reactV");
var List$3 = require("reactX");
var Local = require("./local");
var WebpackConfigJs = require("../../../webpack.config.js");
var WebpackMiddlewareConfigJs = require("../../../webpack.middleware.config.js");

var configx = WebpackConfigJs;

var WebpackConfig = {
  configx: configx
};

var configx$1 = WebpackMiddlewareConfigJs;

var WebpackDevMiddlewareConfig = {
  configx: configx$1
};

function configX(prim) {
  return WebpackMiddlewareConfigJs.configX();
}

function configX$1(prim) {
  return WebpackConfigJs.configX();
}

var U = {
  configX: configX$1
};

var A = { };

var B = { };

function f(param) {
  return /* tuple */[
          (function (prim) {
              List$3.ff();
              
            }),
          (function (prim) {
              List$3.ff2();
              
            }),
          (function (prim) {
              List$2.ff();
              
            }),
          (function (prim) {
              List$2.ff2();
              
            })
        ];
}

List$1.xx();

List.length(/* :: */[
      1,
      /* :: */[
        2,
        /* [] */0
      ]
    ]);

List.length(/* [] */0);

function ff(prim) {
  return Local.ff();
}

exports.WebpackConfig = WebpackConfig;
exports.WebpackDevMiddlewareConfig = WebpackDevMiddlewareConfig;
exports.configX = configX;
exports.U = U;
exports.A = A;
exports.B = B;
exports.f = f;
exports.ff = ff;
/* configx Not a pure module */

File: jscomp/test/webpack_config.ml
module type Config = sig
  val configx : Js.Json.t
end
module WebpackConfig: Config = struct
  external configx : Js.Json.t = "../../../webpack.config.js" [@@bs.module]
end

module WebpackDevMiddlewareConfig: Config = struct
  external configx : Js.Json.t = "../../../webpack.middleware.config.js" [@@bs.module]
end

external configX : unit ->  Js.Json.t = ""
[@@bs.module  "../../../webpack.middleware.config.js"]
[@@bs.val]

let configX = configX
module U  : sig 
  val configX : unit -> Js.Json.t 
end = struct 
    external configX : unit -> Js.Json.t =  ""
    [@@bs.module "../../../webpack.config.js" ]
    [@@bs.val]
end
 external hey : unit -> unit  = "xx" [@@bs.module "List"] 
module A = struct 
  external ff : unit -> unit = "" [@@bs.module "reactX", "List" ]
  external ff2 : unit -> unit = "" [@@bs.module "reactX", "List" ]
end
module B = struct 
  external ff : unit -> unit = "" [@@bs.module "reactV", "List"]
  external ff2 : unit -> unit = "" [@@bs.module "reactV", "List"]
end

let f ()   = A.ff , A.ff2,  B.ff, B.ff2
 ;; hey () 

 ;; List.length [1;2] , List.length []

type t 
external ff : unit -> t = "" [@@bs.module "./local"]